communication/src/main/java/org/jboss/da/communication/pom/PomAnalyzer.java

```diff
@@ -0,0 +1,246 @@
+import org.commonjava.cartographer.graph.discover.DiscoveryResult;
+import org.commonjava.maven.atlas.graph.rel.ProjectRelationship;
+import org.commonjava.maven.atlas.ident.ref.ProjectVersionRef;
+import org.commonjava.maven.galley.TransferException;
+import org.commonjava.maven.galley.maven.GalleyMavenException;
+import org.commonjava.maven.galley.maven.model.view.MavenPomView;
+import org.commonjava.maven.galley.maven.parse.MavenPomReader;
+import org.commonjava.maven.galley.maven.parse.PomPeek;
+import org.commonjava.maven.galley.maven.util.ArtifactPathUtils;
+import org.commonjava.maven.galley.model.Location;
+import org.commonjava.maven.galley.model.SimpleLocation;
+import org.jboss.da.communication.aprox.model.GAVDependencyTree;
+import org.jboss.da.communication.model.GAV;
+import org.jboss.da.communication.pom.model.MavenProject;
+import org.jboss.da.communication.pom.model.MavenRepository;
+import org.jboss.da.communication.pom.qualifier.DACartographerCore;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.file.Files;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@ApplicationScoped
+public class PomAnalyzer {
+
+    @Inject
+    private PomReader pomReader;
+
+    @Inject
+    @DACartographerCore
+    private CartographerCore carto;
+
+    public GAVDependencyTree readRelationships(File pomRepoDir, File pomPath)
+            throws PomAnalysisException {
+
+        try {
+            GAVDependencyTree root = null;
+
+            File tempDir = Files.createTempDirectory("deps").toFile();
+
+            Map<GAV, GAVDependencyTree> gavDependencyTreeMap = new HashMap<>();
+
+            try {
+                // map of file and ProjectVersionRefs that represent all the pom.xmls found in the pomRepoDir directory
+                Map<File, ProjectVersionRef> projectVersionRefs = getProjectVersionRefs(tempDir,
+                        findAllPomFiles(pomRepoDir));
+
+                // for each pom.xml
+                for (File path : projectVersionRefs.keySet()) {
+                    ProjectVersionRef ref = projectVersionRefs.get(path);
+
+                    // find all the dependencies / plugins / parents / plugins etc for each pom.xml
+                    Set<ProjectRelationship<?, ?>> relationships = getDiscoveryResult(tempDir,
+                            pomRepoDir, ref).getAcceptedRelationships();
+
+                    boolean rootRelationship = false;
+
+                    // need to use canonical path to get better way of knowing if 2 paths are the same
+                    if (pomPath.getCanonicalPath().equals(path.getCanonicalPath())) {
```

Comment:
pomPath will be the folder where the root pom.xml is located.

I've done it so that in the request for scm endpoint you can send as the pomPath the empty string that would mean:
- get the artifact in the top-level pom.xml

Or you can specify the path to the pom.xml to be considered root:
- e.g pomPath = "application" <-- for dependency-analysis